app/controllers/api/my_request/requests_controller.rb

```diff
@@ -34,6 +34,7 @@ def create
   def update
     request = Request.find(update_params[:id])
     request.is_requested_by?(current_user) && request.send("#{update_params[:activity]}!".to_sym)
+    request.helper.reward_karma_points(request.reward)
```

Comment:
this should probably be called from the request model, according to latest talk with Thomas?

Reply:
Yeah, was torn between doing it here and doing it from where the request is set as completed in the model. I thought the flow was more readable here, but being readable is perhaps none of the concern of a controller

Reply:
Yes I thought the same about the other change. But I think it should have an after_update or something in the controller, that calls the reward_karma_points in the model

Reply:
I now make the call to the Request that then calls on the user method.
Making a hook would make it a lot more complicated, since update to active is also done by a hook, and I don't want to have special logic and conditionals to not trigger after that hook is called, just to be able to move it.

Reply:
Yes, the thing is that we don't want to depend on the controller for the transaction to happen. I think it's just a matter of adding this to the request model?
```
after_update :reward_karma_points


def reward_karma_points
  if completed
    reward....
```

Reply:
Yeah, that works.